### PR TITLE
feature(odata-service-inquirer): Adds service url prompting

### PR DESCRIPTION
**Pull request by IainSAP**
https://github.com/SAP/open-ux-tools/issues/1987

-  Adds service url support to `@sap-ux/odata-service-inquirer`
- Replaces `@sap/wing-service-explorer` with `fast-xml-parser` for metadata xml validation
- Adds tests to cover

### CODE REVIEW

File: packages/odata-service-inquirer/src/prompts/datasources/service-url/questions.ts [hunk 1]
```diff
@@ -125,24 +125,32 @@ function getCliIgnoreCertValidatePrompt(
...
+                                serviceUrl,
+                                connectValidator,
+                                requiredVersion,
+                                true
+                            );
+                            if (validService !== true) {
+                                throw new Error(t('errors.exitingGeneration', { exitReason: validService.toString() }));
+                            }
                         }
+                        return true;
```
Comment (IainSAP) [inline packages/odata-service-inquirer/src/prompts/datasources/service-url/questions.ts:150]: This when condition should never return true!

**Review by cianmSAP — APPROVED** [review]
Updates to add service url prompting are clear
Coverage above 90
Changeset ✅

File: packages/odata-service-inquirer/src/prompts/datasources/service-url/validators.ts [hunk 2]
```diff
@@ -0,0 +1,85 @@
+import { createForAbap, type AxiosRequestConfig, type ODataService, type ODataVersion } from '@sap-ux/axios-extension';
+import type { OdataVersion } from '@sap-ux/odata-service-writer';
```
Comment (unseen1980) [inline packages/odata-service-inquirer/src/prompts/datasources/service-url/validators.ts:2]: There is an OdataVersion import in line 1 ?

File: packages/odata-service-inquirer/src/prompts/datasources/service-url/validators.ts [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+            PromptState.odataService.annotations = annotations;
+        } catch (err) {
+            LoggerHelper.logger.info(t('prompts.validationMessages.annotationsNotFound'));
+        }
+        return true;
+    } catch (error) {
+        delete PromptState.odataService.metadata;
+        // Provide a more specific error message if the metadata service URL is not found
+        if (ErrorHandler.getErrorType(error) === ERROR_TYPE.NOT_FOUND) {
+            // No metdata implies not a valid odata service
```
Comment (unseen1980) [inline packages/odata-service-inquirer/src/prompts/datasources/service-url/validators.ts:78]: typo

File: packages/odata-service-inquirer/src/prompts/datasources/service-url/connectionValidator.ts [hunk 4]
```diff
@@ -0,0 +1,280 @@
...
+ */
+interface Validity {
+    urlFormat?: boolean;
+    reachable?: boolean;
+    authRequired?: boolean;
+    authenticated?: boolean;
+    canSkipCertError?: boolean;
+}
+
+// Cert errors than may be ignored by prompt user
```
Comment (unseen1980) [inline packages/odata-service-inquirer/src/prompts/datasources/service-url/connectionValidator.ts:22]: Typo?

**Review by unseen1980 — APPROVED** [review]
Code looks good, but I haven't tested it locally. Thanks

File: packages/odata-service-inquirer/src/prompts/datasources/service-url/validators.ts (hunk 2 above)
Comment (IainSAP) [reply packages/odata-service-inquirer/src/prompts/datasources/service-url/validators.ts:2]: They are different types, unfortunately!

File: packages/odata-service-inquirer/src/prompts/datasources/service-url/connectionValidator.ts (hunk 4 above)
Comment (IainSAP) [reply packages/odata-service-inquirer/src/prompts/datasources/service-url/connectionValidator.ts:22]: Thx! Will fix is pending PR: https://github.com/SAP/open-ux-tools/pull/2036

File: packages/odata-service-inquirer/src/prompts/datasources/service-url/validators.ts (hunk 3 above)
Comment (IainSAP) [reply packages/odata-service-inquirer/src/prompts/datasources/service-url/validators.ts:78]: Thx! Will fix is pending PR: https://github.com/SAP/open-ux-tools/pull/2036

**Pull request merged**